File: backend/internal/model/gran_premio.go
package model

import "time"

// EstadoGP representa el estado de un Gran Premio.
type EstadoGP string

const (
	EstadoPendiente  EstadoGP = "pendiente"
	EstadoEnCurso    EstadoGP = "en_curso"
	EstadoCompletado EstadoGP = "completado"
)

// GranPremio representa un GP del calendario de una temporada.
type GranPremio struct {
	ID          int       `json:"id"`
	TemporadaID int       `json:"temporada_id"`
	Nombre      string    `json:"nombre"`
	Circuito    *string   `json:"circuito,omitempty"`
	Pais        *string   `json:"pais,omitempty"`
	Fecha       *time.Time `json:"fecha,omitempty"`
	TieneSprint bool      `json:"tiene_sprint"`
	Estado      EstadoGP  `json:"estado"`
	Orden       int       `json:"orden"`
}

File: frontend/src/types/gran-premio.ts
export type EstadoGP = "pendiente" | "en_curso" | "completado";

export interface GranPremio {
  id: number;
  temporada_id: number;
  nombre: string;
  circuito?: string;
  pais?: string;
  fecha?: string;
  tiene_sprint: boolean;
  estado: EstadoGP;
  orden: number;
}
